.github/workflows/trigger-gitlab-pipeline.yml

```
@@ -5,7 +5,40 @@ on:
       - main  # This ensures the workflow triggers on pushes to the main branch, including merges
 
 jobs:
-  trigger-gitlab:
+  trigger-and-check-gitlab:
     runs-on: ubuntu-latest
     steps:
-      - run: curl --request POST --form token=${{secrets.GITLAB_TOKEN}} --form ref=main "https://gitlab.com/api/v4/projects/35484791/trigger/pipeline"
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - name: Trigger GitLab Pipeline
+        id: trigger
+        run: |
+          response=$(curl --request POST \
+                           --form token=${{ secrets.GITLAB_TRIGGER_TOKEN }} \
+                           --form ref=main \
+                           "https://gitlab.com/api/v4/projects/35484791/trigger/pipeline")
+          echo "::set-output name=pipeline_id::$(echo $response | jq -r '.id')"
+      
+      - name: Poll Pipeline Status
+        run: |
+          status=""
+          while [[ "$status" != "success" && "$status" != "failed" ]]; do
```

```suggestion
          while [[ "$status" != "success" ]]; do
```
we are checking if the status is failed inside the while, so I think it should work without this check here